qa/acceptance/spec/lib/cli/logstash-plugin/install_spec.rb

```diff
@@ -0,0 +1,65 @@
+# encoding: utf-8
+require_relative "../../../spec_helper"
+require "logstash/version"
+require "fileutils"
+
+shared_examples "logstash install" do |logstash|
+  let(:logstash_path) { "/opt/logstash" }
+
+  before(:each) do
+    logstash.install(LOGSTASH_VERSION)
+  end
+
+  after :each do
+    logstash.uninstall
+  end
+
+  describe "on #{name}" do
+    context "with a direct internet connection" do
+      context "when the plugin exist" do
+        context "from a local `.GEM` file" do
+          let(:gem_path) { download_gem("logstash-filter-dns", "2.1.3") }
+          let(:gem_path_on_vagrant) { "/vagrant/vendor/download/logstash-filter-dns-2.1.3.gem" }
+          after(:each) { FileUtils.rm_rf(gem_path) }
+
+          it "successfully install the plugin" do
+            command = run_command_in_path("bin/logstash-plugin install #{gem_path_on_vagrant}", logstash.host)
+            expect(command).to run_successfully_and_output(/Installation successful/)
+            expect(plugin_installed?(logstash.host, "logstash-filter-dns")).to be_truthy
+          end
+        end
+
+        context "when fetching a gem from rubygems" do
+          it "successfully install the plugin" do
+            command = run_command_in_path("bin/logstash-plugin install logstash-filter-dns", logstash.host)
+            expect(command).to run_successfully_and_output(/Installation successful/)
+            expect(plugin_installed?(logstash.host, "logstash-filter-dns")).to be_truthy
+          end
+
+          it "allow to install a specific version" do
+            command = run_command_in_path("bin/logstash-plugin install --version 2.1.2 logstash-filter-dns", logstash.host)
+            expect(command).to run_successfully_and_output(/Installation successful/)
+            expect(plugin_installed?(logstash.host, "logstash-filter-dns", "2.1.2")).to be_truthy
+          end
+        end
+      end
+
+      context "when the plugin doesnt exist" do
+        it "fails to install and report an error" do
+          # LOOK WITH PERE WHY EXITSTATUS IS NIL
```

Comment:
@purbon If you uncomment the check `fail_and_output` that does an explicit check on the status code, the remote command will return `NIL` instead of 1.